File: server/src/models/manifest.rs
#[derive(
    Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, ToSchema, ToResponse, Serialize, Deserialize,
)]
pub struct PackageManifest {
    /// The package name
    pub name: String,

    /// The package author
    pub authors: Vec<String>,

    /// The package version
    pub version: String,

    /// The package description
    pub description: String,

    /// The KubeJS versions this package works on
    pub kubejs: Vec<String>,

    /// The loaders this package works on
    pub loaders: Vec<String>,

    /// This package's dependencies
    pub dependencies: Vec<String>,

    /// This package's incompatibilities
    pub incompatibilities: Vec<String>,
}

File: ui/src/lib/types/manifest.ts
export type ModLoader = "forge" | "fabric" | "quilt" | "neoforge";

export interface PackageManifest {
    /**
     * The package's name.
     */
    name: string;

    /**
     * The package's authors.
     */
    authors: string[];

    /**
     * The package version.
     */
    version: string;

    /**
     * A short description of the package.
     */
    description: string;

    /**
     * A list of game versions this package works on.
     * Can be `["all"]`.
     */
    game_versions: string[];

    /**
     * A list of loaders this package works on.
     * Must NOT be empty.
     */
    loaders: ModLoader[];

    /**
     * A list packages this one depends on.
     * Can be empty.
     */
    dependencies: string[];

    /**
     * A list of packages this one is incompatible with.
     * Can be empty.
     */
    incompatibilities: string[];
}
